content/blog/vscode-esp-idf-v1.8.0/index.md

```diff
@@ -0,0 +1,133 @@
+
+{{< figure
+    default=true
+    src="img/debug-arch.webp"
+    alt=""
+    caption=""
+    >}}
+
+### The actual Eclipse CDT GDB Debug Adapter
+
+Before we have implemented our own [ESP-IDF Debug Adapter](https://github.com/espressif/esp-debug-adapter) in Python. While this debug adapter works, our users have report many issues regarding configuration, responsiveness and runtime errors. In our search for solution we encountered that the Eclipse CDT Cloud team have published a [CDT-GDB-Adapter](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter) which is written in TypeScript and is using [NPM serialport package](https://www.npmjs.com/package/serialport) underneath. We added a few changes in order to make it work with our tools but it is mostly the same as the original.
+
+To use the Eclipse CDT GDB Adapter in an ESP-IDF project, add the following configuration in `.vscode/launch.json`:
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdbtarget",
+      "request": "attach",
+      "name": "Eclipse CDT GDB Adapter"
+    }
+  ]
+}
+```
+
+To start debugging the project, do the following:
+* Build (command palette: **ESP-IDF: Build your Project**)
+* Flash (command palette: **ESP-IDF: Flash your Project**)
+* In the `Run` menu, press `Start debugging`.
+
+You can also check how to use the new debugger by following the [debug tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/debugging.md).
+
+The next figure shows an active debug session with the Eclipse CDT GDB Adapter in Visual Studio Code. You can see the list of variables, call stack, breakpoints, debug console and debug controls such as step in, step out, etc.
+
+{{< figure
+    default=true
+    src="img/debug-example.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Advanced configuration of Eclipse CDT Debug Adapter
+
+There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).
+
+## ESP-IDF version switcher
+
+Whenever you install ESP-IDF in your system using the command `ESP-IDF: Configure ESP-IDF Extension`, the installations are saved in our extension as global state IDF setup. Now you can use `ESP-IDF: Select Current ESP-IDF Version` to set the ESP-IDF version you want to use in your current project (_workspace_ folder in VS Code terminology). If you want to delete these IDF setups from the extension global state, you can use `ESP-IDF: Clear Saved ESP-IDF Setups`.
+
+{{< figure
+    default=true
+    src="img/idf-version-list.webp"
+    alt=""
+    caption=""
+    >}}
+
+The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:
+
+{{< figure
+    default=true
+    src="img/idf-version-status-bar.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Use code navigation without full project build
```

Comment:
Can you please add the information on why the user might not want to fully build the project? In which case it is good to do or what are the benefits?

You can either add it to the article yourself or drop some ideas in a comment and I will what to add to the article myself.

This is a question from a [previous comment](https://github.com/espressif/developer-portal/pull/62#discussion_r1685976723).

Reply:
Basically because doing a full build will require longer time and this reconfigure task is the first step to build but it is faster. On any regular build this command is actually run first to configure project and then ninja is used as build tool run the rest of the build as described in here: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#using-cmake-directly

Reply:
Thanks! I will make respective updates in the following comments.